geode-protobuf-messages/src/main/proto/v1/function_API.proto

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+syntax = "proto3";
+package org.apache.geode.internal.protocol.protobuf.v1;
+
+import "v1/basicTypes.proto";
+
+message ExecuteFunctionOnRegionRequest {
+    string functionID = 1;
+    string region = 2;
+    EncodedValue arguments = 3;
```

Comment:
Should `arguments` also be repeated?

Reply:
Second Sarge's comment.

Reply:
Well, this is a weird one. I think you've found an issue with the API representation here: the default collector returns a `List<ResultType>` but custom collectors could return anything; and functions themselves get a single argument named `arguments` (see `FunctionContext.getArguments();`). The state here is the same as that, but not in a good way.